### PR TITLE
Adds a copy-url-button component

### PR DESCRIPTION
**Pull request by mynameistechno**
This PR:
* Adds a copy-url-button component that leverages `react-copy-to-clipboard`
* Adds the button to components list, inspect, and curate pages (not sure we want them on all, but can easily change)
* A couple UX clean up items (just experimenting, I can revert/change):
  * Groups the icon buttons in the component list
  * Moves the badge over the left so it doesn't shift the buttons to the left after it renders (this is temporary, long term we can do better)
  * Adds bootstrap's "well" styling around the filter boxes

image

On the Inspect/Curate pages I tried a more subtle "button link" style floated to the right. I didn't want the user to confuse it with a search button (but it looks good as a button and I'm open to changing it or removing altogether):

image

### CODE REVIEW

File: src/components/CopyUrlButton.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  renderTooltip() {
+    return (
+      <Tooltip id="tooltip">{this.state.copied ? "Copied!" : "Copy URL to clipboard"}</Tooltip>
+    )
+  }
+
+  render() {
+    const {path, bsStyle = 'link', className} = this.props
```
Comment (jeffmcaffer) [inline src/components/CopyUrlButton.js]: suggest adding the `bsStyle` default value as a default prop to make it clear to component users.

File: src/components/CopyUrlButton.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Button, Tooltip, OverlayTrigger } from 'react-bootstrap'
+import { CopyToClipboard } from 'react-copy-to-clipboard'
+import FontAwesome from 'react-fontawesome'
+
+export default class CopyUrlButton extends Component {
+
+  static propTypes = {
+    path: PropTypes.string
```
Comment (jeffmcaffer) [inline src/components/CopyUrlButton.js]: should have bsStyle and className as declared props (with defaults ideally)

File: src/components/CopyUrlButton.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  onClick(event) {
+    event.stopPropagation()
+  }
+
+  onCopy() {
+    this.state.timeoutId && clearTimeout(this.state.timeoutId)
+    const timeoutId = setTimeout(this.didCopy, 2000)
+    this.setState({...this.state, copied: true, timeoutId})
```
Comment (jeffmcaffer) [inline src/components/CopyUrlButton.js]: formatting nit wrt space after { and before } for object literals.

File: src/components/PageComponentDetails.js [hunk 4]
```diff
@@ -118,10 +118,13 @@ class PageInspect extends Component {
     const { filterOptions, filterValue, definition, curation, harvest } = this.props
     return (
       <Grid className='main-container'>
-        <Row className="show-grid spacer">
-          <Col md={10} mdOffset={1}>
+        <Row className="show-grid well">
```
Comment (jeffmcaffer) [inline src/components/PageComponentDetails.js]: not keen on this being a well. The design is meant to be super clean without boxes, borders, grid lines, ...  Is this needed?

File: src/components/CopyUrlButton.js (hunk 3 above)
Comment (mynameistechno) [reply src/components/CopyUrlButton.js]: Prettier wherefore art thou 😜

File: src/components/CopyUrlButton.js (hunk 2 above)
Comment (mynameistechno) [reply src/components/CopyUrlButton.js]: Sounds good, will define and add defaults for all props.

File: src/components/PageComponentDetails.js (hunk 4 above)
Comment (mynameistechno) [reply src/components/PageComponentDetails.js]: No I can remove, it's just that `spacer` had issues. I'll just add more margin between navbar and content to achieve same effect as `spacer`.

**Comment by mynameistechno** [comment]
Feedback addressed @jeffmcaffer. Also, I removed the `well` class and did some minor cleanup in the components list. The source line/sha was causing wrapping on the buttons on the right, so I added ellipsis text overflow support for the source value so the layout handles large source values more gracefully (and a couple other css fixes):

Before:
image

After:
image

The panel/list expand below needs some work, but that can be tackled separately. E.g. if browser width is narrowed a bit:
image

Created #74 to that effect.

We should prob merge #68 first so I can make sure the filterbar renders well with the facets.

**Pull request merged**